xline/src/server/kv_server.rs

```diff
@@ -198,13 +198,19 @@ where
     }
 
     /// Validate range request before handle
-    fn check_range_request(req: &RangeRequest) -> Result<(), tonic::Status> {
+    fn check_range_request(req: &RangeRequest, current_revision: i64) -> Result<(), tonic::Status> {
         if req.key.is_empty() {
             return Err(tonic::Status::invalid_argument("key is not provided"));
         }
         if !SortOrder::is_valid(req.sort_order) || !SortTarget::is_valid(req.sort_target) {
             return Err(tonic::Status::invalid_argument("invalid sort option"));
         }
+        if req.revision > current_revision {
```

Comment:
Why do we need to check the revision here now when we didn't need to before?

Reply:
it was a todo before
https://github.com/datenlord/Xline/pull/247#discussion_r1190788475